contracts/tests/test_validator_set.py

```diff
@@ -142,6 +142,49 @@ def test_cannot_call_finalize_change(validator_set_contract_session, accounts):
         contract.functions.finalizeChange().transact({"from": accounts[0]})
 
 
+def test_finalize_change_update_history_data_correctly(
+    validator_set_contract_session, accounts, web3
+):
+    assert (
+        validator_set_contract_session.functions.getLengthOfBlockNumbersOfFinalizedChanges().call()
+        == 0
+    )
+
+    new_validator_set = accounts[:2]
+    validator_set_contract_session.functions.testChangeValiatorSet(
+        new_validator_set
+    ).transact()
```

Comment:
We should probably test that nothing has changed at this point.

Reply:
I have added a test case for that. :+1: